redis/src/main/java/org/togglz/redis/RedisStateRepository.java

```diff
@@ -0,0 +1,151 @@
+package org.togglz.redis;
+
+import org.togglz.core.Feature;
+import org.togglz.core.repository.FeatureState;
+import org.togglz.core.repository.StateRepository;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.Protocol;
+
+import java.util.Map;
+
+/**
+ * A state repository which stores the feature state in Redis.
+ * <p>
+ * The class provides a builder which can be used to configure the state repository instance
+ * (e.g. Redis hostname, Redis/Jedis configuration, serialization).
+ * </p>
+ *
+ * @author Cosmin Rentea
+ */
+public class RedisStateRepository implements StateRepository {
+
+    public static final String KEY_PREFIX = "togglz:";
+    public static final String ENABLED_FIELD = "enabled";
+    public static final String STRATEGY_FIELD = "strategy";
+    public static final String PARAMETER_PREFIX = "parameter:";
+    public static final int PARAMETER_PREFIX_LENGTH = PARAMETER_PREFIX.length();
+
+    protected final JedisPool jedisPool;
```

Comment:
I'm starting to ask myself, whether it wouldn't be much simpler if the user could just provide a `JedisPool` directly via the builder. Then we could remove `jedisPoolConfig` and `hostname` which makes the repository much simpler. For the user this could also be much simpler, because if he already has a `JedisPool` in his application, he could simply use this one as a shared pool between Togglz and his app. What do you think?